### PR TITLE
refactor: clean up files in matching-service

### PR DESCRIPTION
**Pull request by tituschewxj**

### CODE REVIEW

**Review by bensohh — DISMISSED** [review]
LGTM 👍

File: apps/matching-service/handlers/websocket.go [hunk 1]
```diff
@@ -170,7 +168,10 @@ func waitForResult(ws *websocket.Conn, ctx, timeoutCtx, matchCtx context.Context
 		}
 		log.Println("Match found for user: " + username)
 		// Notify the users about the match
-		notifyMatch(result.User, result.MatchedUser, result)
+		notifyMatches(result.User, result.MatchedUser, result)
+
```
Comment (bensohh) [inline apps/matching-service/handlers/websocket.go:172]: Previously:
I think that after notifying matches, we should close the matchChannel for the matchedUser so he/she does not end up having a timeout. Previously, what I did was to call the cancelFunc() to cancel the matchCtx for both users which prevented the match_found then timeout error. But maybe there is a better way to do this?

One of the user will enter the case result, ok := <-matchFoundChan while the other wouldn't. If not what we can do it to notify both users in the PerformMatching

**Comment by bensohh** [comment]
I added a quick modification to the PerformMatching function to notify both matching channels and NotifyUser function. In that case perhaps we can look to removing matchCtx if not in use?

**Review by bensohh — APPROVED** [review]
LGTM

File: apps/matching-service/handlers/websocket.go (hunk 1 above)
Comment (tituschewxj) [reply apps/matching-service/handlers/websocket.go:172]: >we should close the matchChannel for the matchedUser so he/she does not end up having a timeout

Agreed.

>what we can do it to notify both users in the PerformMatching

This works for now.

**Pull request merged**